packages/medusa/src/api/admin/products/import/route.ts

```diff
@@ -2,27 +2,21 @@ import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "@medusajs/framework/http"
-import { HttpTypes } from "@medusajs/framework/types"
-import { MedusaError } from "@medusajs/framework/utils"
+import type { HttpTypes } from "@medusajs/framework/types"
 import { importProductsWorkflow } from "@medusajs/core-flows"
+import type { AdminImportProductsType } from "../validators"
 
 export const POST = async (
-  req: AuthenticatedMedusaRequest<HttpTypes.AdminImportProductRequest>,
+  req: AuthenticatedMedusaRequest<AdminImportProductsType>,
   res: MedusaResponse<HttpTypes.AdminImportProductResponse>
 ) => {
-  const input = req.file as Express.Multer.File
-
-  if (!input) {
-    throw new MedusaError(
-      MedusaError.Types.INVALID_DATA,
-      "No file was uploaded for importing"
-    )
-  }
+  const fileProvider = req.scope.resolve("file")
+  const file = await fileProvider.getAsBuffer(req.body.filename)
```

Comment:
NIT: Maybe it's better to call it `fileId` or `fileKey` so it's clear that it is not a user-friendly name but the identifier in storage

Reply:
Yeah, the `fileKey` makes more sense.

Reply:
todo: Use `validatedBody`